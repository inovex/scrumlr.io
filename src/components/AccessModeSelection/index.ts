import AccessModeSelection from './AccessModeSelection';
export default AccessModeSelection;
export * from './AccessModeSelection';
