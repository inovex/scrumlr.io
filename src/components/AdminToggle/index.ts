export * from './AdminToggle';

import AdminToggle from './AdminToggle';
export default AdminToggle;
