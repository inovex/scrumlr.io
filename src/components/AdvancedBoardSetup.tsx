import React from 'react';
import Typography from '@material-ui/core/Typography';
import FormGroup from '@material-ui/core/FormGroup';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import { useTranslation } from 'react-i18next';

export interface AdvancedBoardSetupProps {
  admissionControl: boolean;
  encryptedData: boolean;
  onSetAdmissionControl: (enabled: boolean) => void;
  onSetEncryptedData: (enabled: boolean) => void;
}

export const AdvancedBoardSetup: React.FC<AdvancedBoardSetupProps> = ({ admissionControl, encryptedData, onSetAdmissionControl, onSetEncryptedData }) => {
  const { t } = useTranslation('advancedBoardSetup');

  return (
    <>
      <ExpansionPanel>
        <ExpansionPanelSummary>
          <Typography>{t('heading')}</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <div>
            <Typography>{t('helper')}</Typography>
            <FormGroup>
              <FormControlLabel
                onClick={(event) => event.stopPropagation()}
                onFocus={(event) => event.stopPropagation()}
                control={
                  <Switch
                    checked={admissionControl}
                    onChange={(event) => {
                      onSetAdmissionControl(event.target.checked);
                    }}
                  />
                }
                label={t('admissionControl.heading')}
              />
              <Typography color="textSecondary">{t('admissionControl.helper')}</Typography>

              <FormControlLabel
                onClick={(event) => event.stopPropagation()}
                onFocus={(event) => event.stopPropagation()}
                control={<Switch checked={encryptedData} onChange={(event) => onSetEncryptedData(event.target.checked)} disabled={!admissionControl} />}
                label={t('encryptData.heading')}
              />
              <Typography color="textSecondary">{t('encryptData.helper')}</Typography>
            </FormGroup>
          </div>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </>
  );
};

export default AdvancedBoardSetup;
