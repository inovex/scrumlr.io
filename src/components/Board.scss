@import "common";

.board {
  --n: 1;

  display:grid;
  grid-auto-flow: column;
  grid-auto-columns: calc(100%/min(var(--real-columns), var(--n)));
  overflow-x:auto;
  scroll-snap-type: x mandatory;

  height: 100vh;
}

@for $i from 1 through 12 {
  @media screen and (min-width: $i*$min-column-width) {
    .board {
      --n: #{$i};
    }
  }
}