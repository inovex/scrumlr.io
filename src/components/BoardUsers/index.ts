export * from "./BoardUsers";
export * from "./UserAvatar";
