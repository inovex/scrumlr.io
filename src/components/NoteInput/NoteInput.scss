@import "constants/style";

.note-input__icon {
  width: 14px;
  height: 14px;
  fill: var(--accent-color);
  align-self: center;
}
