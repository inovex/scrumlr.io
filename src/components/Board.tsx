import React from 'react';
import './Board.scss';
import {getColorClassName} from "../constants/colors";

const Board = ({ children }: any) => {
    const colors = React.Children.map(children, (child) => child.props.color);

    return (
        <>
            <style>
                {`.board { --board__columns: ${React.Children.count(children)} }`}
            </style>
            <div className="board">
                <div className={`board__spacer-left ${getColorClassName(colors[0])}`} />
                    {children}
                <div className={`board__spacer-right ${getColorClassName(colors[colors.length - 1])}`} />
            </div>
        </>
    )
};

export default Board;