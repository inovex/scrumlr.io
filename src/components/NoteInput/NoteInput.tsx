import React from "react";
import "./NoteInput.scss";
import {ReactComponent as PlusIcon} from "assets/icon-add.svg";
import store from "store";
import {ActionFactory} from "store/action";
import {InputAdornment, createStyles, withStyles, FilledInput} from "@material-ui/core";

export interface NoteInputProps {
  columnId: string;
}

const CustomInput = withStyles(
  () =>
    createStyles({
      root: {
        borderRadius: 20,
        height: 40,
        width: "100%",
        backgroundColor: "#EDEFF2",
        marginTop: 16,
        marginBottom: 32,
        "&.MuiFilledInput-underline::before , &.MuiFilledInput-underline::after": {
          display: "none",
        },
        "&:hover , &.Mui-focused": {
          backgroundColor: "white",
          boxShadow: "0 6px 9px 0 rgba(0,87,255,0.16)",
        },
      },
      input: {
        color: "black",
        fontSize: 14,
        fontWeight: "bold",
        lineHeight: 24,
        letterSpacing: 0.25,
        padding: "10px 20px 10px 20px",
        // Use the system font instead of the default Roboto font.
        fontFamily: ["Raleway", "sans-serif"].join(","),
        "& .MuiFilledInput-underline:before": {
          borderBottomColor: "green",
          borderBottom: 0,
          position: "relative",
          left: 20,
        },
      },
      "@media (prefers-color-scheme: dark)": {
        root: {
          backgroundColor: "#4C5566",
          "&:hover , &.Mui-focused": {
            backgroundColor: "#4C5566",
            boxShadow: "0 6px 9px 0 #232323",
          },
        },
        input: {
          color: "white",
        },
      },
    }),
  {name: "NoteInput"}
)(FilledInput);

const NoteInput = ({columnId}: NoteInputProps) => {
  const [value, setValue] = React.useState("");

  function handleChangeNotetext(e: React.ChangeEvent<HTMLInputElement>) {
    setValue(e.target.value);
  }
  const onAddNote = () => {
    if (value) {
      store.dispatch(ActionFactory.addNote(columnId!, value));
      setValue("");
    }
  };
  return (
    <CustomInput
      placeholder="Add your note..."
      type="text"
      value={value}
      onChange={handleChangeNotetext}
      onKeyDown={(e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.key === "Enter") {
          onAddNote();
        }
      }}
      endAdornment={
        <InputAdornment position="end">
          <PlusIcon onClick={onAddNote} className="note-input__icon" />
        </InputAdornment>
      }
    />
  );
};
export default NoteInput;
