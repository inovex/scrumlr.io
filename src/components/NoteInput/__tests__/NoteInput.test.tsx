import {fireEvent, render} from "@testing-library/react";
import {NoteInput} from "components/NoteInput";
import {I18nextProvider} from "react-i18next";
import i18nTest from "i18nTest";
import {Provider} from "react-redux";
import getTestStore from "utils/test/getTestStore";

const createNoteInput = (columnId: string, maxNoteLength: number) => (
  <I18nextProvider i18n={i18nTest}>
    <Provider store={getTestStore()}>
      <NoteInput columnId={columnId} maxNoteLength={maxNoteLength} columnIndex={1} columnIsVisible toggleColumnVisibility={() => undefined} />
    </Provider>
  </I18nextProvider>
);

describe("Note Input", () => {
  beforeEach(() => {
    window.IntersectionObserver = jest.fn(
      () =>
        ({
          observe: jest.fn(),
          disconnect: jest.fn(),
        }) as unknown as IntersectionObserver
    );
  });

  test("should render correctly", () => {
    const {container} = render(createNoteInput("TestID", 1024));
    expect(container.firstChild).toMatchSnapshot();
  });

  test("note length", () => {
    const {container} = render(createNoteInput("TestID", 5));

    // less works as expected
    fireEvent.change(container.querySelector(".note-input__input")!, {target: {value: "1234"}});
    fireEvent.keyDown(container.querySelector(".note-input__input")!, {key: "Enter", code: "Enter", charCode: 13});

    // Exact works as expected
    fireEvent.change(container.querySelector(".note-input__input")!, {target: {value: "12345"}});
    fireEvent.keyDown(container.querySelector(".note-input__input")!, {key: "Enter", code: "Enter", charCode: 13});

    // More than the limit works as expected
    fireEvent.change(container.querySelector(".note-input__input")!, {target: {value: "123456"}});
    fireEvent.keyDown(container.querySelector(".note-input__input")!, {key: "Enter", code: "Enter", charCode: 13});
  });
});
