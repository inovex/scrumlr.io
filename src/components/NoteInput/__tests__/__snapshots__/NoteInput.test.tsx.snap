// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Input should render correctly 1`] = `
<form
  class="note-input"
>
  <textarea
    class="note-input__input"
    id="note-input-TestID"
    maxlength="1024"
    placeholder="Add your note..."
    rows="1"
    style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
  />
  <div
    class="note-input__emoji-suggestions"
  />
  <button
    aria-label="Create note"
    class="note-input__add-button"
    tabindex="-1"
    type="submit"
  >
    <svg
      class="note-input__icon--add"
    >
      icon-add.svg
    </svg>
  </button>
</form>
`;
